core/state/cache.go

```diff
@@ -0,0 +1,145 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	// DefaultMaxLayers is the default maximum number of layers to keep in the cache
+	DefaultMaxLayers = 128
+)
+
+type diffCache struct {
+	storageDiffs      map[felt.Felt]map[felt.Felt]*felt.Felt // addr -> {key -> value, ...}
+	nonces            map[felt.Felt]*felt.Felt               // addr -> nonce
+	deployedContracts map[felt.Felt]*felt.Felt               // addr -> class hash
+}
+
+type stateCache struct {
+	diffs      map[felt.Felt]*diffCache // state root -> state diff
+	links      map[felt.Felt]felt.Felt  // child -> parent
+	oldestRoot felt.Felt                // root of the oldest layer in the cache
+}
+
+func newStateCache() *stateCache {
+	return &stateCache{
+		diffs: make(map[felt.Felt]*diffCache),
+		links: make(map[felt.Felt]felt.Felt),
```

Comment:
Also, for future references just as slices can be created with a fixed size, so can maps. Not something necessary to apply now, since we might change the implementation

```suggestion
		diffs: make(map[felt.Felt]*diffCache, constant),
		links: make(map[felt.Felt]felt.Felt, constant),
```

Reply:
added